Adds cat data frame analytics API spec

szabosteve opened the pull request:
## Overview

This PR adds missing fields to the [cat data frame analytics API](https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-dfanalytics.html).

swallez commented on specification/cat/_types/CatBase.ts:
```diff
@@ -28,3 +28,91 @@ import { RequestBase } from '@_types/Base'
+   * Contains messages relating to the selection of a node.
+   * @aliases ae
+   */
+  assignment_explanation = 0,
```
You can drop all the ` = <number>`, it is not needed and just `assignment_explanation,` is fine.

There is still a lot of occurrences of "numbered enumerations" in the TypeScript files that were output by the script that created the initial version of the spec from the Nest client code. We have to clean things up and remove them all, and can happily avoid adding them in new specifications.

delvedor reviewed (commented):
The changes looks ok, but please take a look at https://github.com/elastic/elasticsearch-specification/pull/1351 first and merge it in your branch once that one is merged :)

szabosteve commented:
@swallez @delvedor Thank you for your reviews. I addressed your comments. However, I close this PR in favor of https://github.com/elastic/elasticsearch-specification/pull/1355 as the conflicts became hardly manageable in this pull request.
